File: LeetCode/121-BestTimeToBuyAndSellStocks.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        buy = float('inf')
        maxProfit = 0

        for price in prices:
            buy = min(buy, price)
            maxProfit = max(maxProfit, price - buy)

        return maxProfit

File: LeetCode/Solutions/Java/121-BestTimeToBuyAndSellStocks.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyDay = 0;
        int sellDay = buyDay + 1;
        int profit = 0;
        int maxProfit = 0;


        while(sellDay < prices.length) {
            profit = prices[sellDay] - prices[buyDay];

            if(profit > maxProfit) {
                maxProfit = profit;
            } else if (profit < 0) {
                buyDay = sellDay;
            }

            sellDay++;
        }

        return maxProfit;
    }
}